Tests/ConduitTests/Networking/Data/DataDownloaderTests.swift

```diff
@@ -0,0 +1,179 @@
+            sut.downloadData(for: dataRequest) { _ in }
+        }
+        XCTAssert(monitoringSessionClient.numRequestsSent == 1)
+    }
+
+    func testMarksDatasAsCachedAfterDownloaded() throws {
+        let attemptedAllDataRetrievalsExpectation = expectation(description: "attempted all data retrievals")
+
+        let sut = DataDownloader(cache: AutoPurgingURLDataCache())
+        let url = try URL(absoluteString: "https://httpbin.org/image/svg")
+        let dataRequest = URLRequest(url: url)
+
+        sut.downloadData(for: dataRequest) { response in
+            XCTAssert(response.isFromCache == false)
+
+            let dispatchGroup = DispatchGroup()
+
+            for _ in 0..<100 {
+                dispatchGroup.enter()
+                DispatchQueue.global().async {
+                    sut.downloadData(for: dataRequest) { response in
+                        XCTAssert(response.isFromCache == true)
+                        dispatchGroup.leave()
+                    }
+                }
+            }
+            dispatchGroup.notify(queue: DispatchQueue.main) {
+                attemptedAllDataRetrievalsExpectation.fulfill()
+            }
+        }
+
+        waitForExpectations(timeout: 5)
+    }
+
+    func testHandlesSimultaneousRequestsForDifferentDatas() {
+        let dataURLs = (0..<10).compactMap {
+            URL(string: "https://httpbin.org/image/svg?id=\($0)")
+        }
+
+        let sut = DataDownloader(cache: AutoPurgingURLDataCache())
+        let fetchedAllDatasExpectation = expectation(description: "fetched all data")
+        let dispatchGroup = DispatchGroup()
+        for url in dataURLs {
+            dispatchGroup.enter()
+            sut.downloadData(for: URLRequest(url: url)) { response in
+                XCTAssertNotNil(response.data)
+                XCTAssertNil(response.error)
+                XCTAssertFalse(response.isFromCache)
+                XCTAssertEqual(response.urlResponse?.statusCode, 200)
+                dispatchGroup.leave()
+            }
+        }
+
+        dispatchGroup.notify(queue: DispatchQueue.main) {
+            fetchedAllDatasExpectation.fulfill()
+        }
+
+        waitForExpectations(timeout: 5)
+    }
+
+    func testPersistsWhileOperationsAreRunning() throws {
+        let dataDownloadedExpectation = expectation(description: "data downloaded")
+        var sut: DataDownloader? = DataDownloader(cache: AutoPurgingURLDataCache())
+        let url = try URL(absoluteString: "https://httpbin.org/image/svg")
+        let dataRequest = URLRequest(url: url)
+
+        weak var weakDataDownloader = sut
+        sut?.downloadData(for: dataRequest) { _ in
+            DispatchQueue.global().asyncAfter(deadline: .now() + 0.5) {
+                XCTAssert(weakDataDownloader == nil)
+                dataDownloadedExpectation.fulfill()
+            }
+        }
+        sut = nil
+        XCTAssert(weakDataDownloader != nil)
+
+        waitForExpectations(timeout: 5)
+    }
+
```

Comment:
Looks great, I'd just recommend adding the `"GIVEN, WHEN, THEN"` statements to the above tests as well.